modules/core/src/main/scala/gem/Location.scala

```diff
@@ -0,0 +1,73 @@
+package gem
+
+import scalaz._, Scalaz._
+import scalaz.Ordering.{EQ, GT, LT}
+
+/** A sortable value used to indicate relative positions of a set of associated
+  * elements.  `Location`s may be thought of as lists of arbitrary integers
+  * which are be compared element by element such that the first pair of values
+  * that differ determine the ordering of the `Location`s as a whole.  If one
+  * `Location` is a proper prefex of another, then it sorts ahead of the other
+  * `Location`.
+  */
+sealed abstract case class Location(toNel: NonEmptyList[Int]) {
```

Comment:
You could demand `Foldable1` for methods that need the non-emptiness. It's really not a big deal. I suggest you merge it as-is and we can make it generic later if we feel like it.